cmd/goal/box.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/base64"
+
+	"github.com/spf13/cobra"
+)
+
+var boxName string
+
+func init() {
+	appCmd.AddCommand(appBoxCmd)
+
+	appBoxCmd.AddCommand(appBoxInfoCmd)
+	appBoxCmd.AddCommand(appBoxListCmd)
+	appBoxCmd.PersistentFlags().Uint64Var(&appIdx, "app-id", 0, "Application ID")
+	appBoxCmd.MarkFlagRequired("app-id")
+
+	appBoxInfoCmd.Flags().StringVarP(&boxName, "name", "n", "", "Application box name. Use the same form as app-arg to name the box.")
+	appBoxInfoCmd.MarkFlagRequired("name")
+}
+
+var appBoxCmd = &cobra.Command{
+	Use:   "box",
+	Short: "Read application box data",
+	Args:  cobra.ArbitraryArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		// If no arguments passed, we should fallback to help
+		cmd.HelpFunc()(cmd, args)
+	},
+}
+
+var appBoxInfoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Retrieve information about an application box.",
+	Long:  "Retrieve information about an application box. The returned box name and value are base64 encoded.",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		_, client := getDataDirAndClient()
```

Comment:
Adding more coverage would be great. I'm not sure we need 100% coverage, but see which of these paths give the greatest "bang for the buck" when adding an error scenario or two to the e2e test.

Reply:
Actually, I'm starting to suspect that the coverage tool isn't working properly for e2e tests. If you agree, just ignore my previous comment.

Reply:
Correct, the coverage tool only knows if we call it from a Go unit test.

Reply:
coverage test never know what's going on for e2e test, e.g., test for REST endpoint